rllib/core/testing/tf/bc_optimizer.py

```diff
@@ -0,0 +1,37 @@
+import tensorflow as tf
+from typing import Any, Mapping
+
+from ray.rllib.core.optim.rl_optimizer import RLOptimizer
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.nested_dict import NestedDict
+
+
+class BCTFOptimizer(RLOptimizer):
+    def __init__(self, module, config):
+        super().__init__(module, config)
+
+    def compute_loss(self, batch: NestedDict, fwd_out) -> Mapping[str, Any]:
+        """Compute a loss"""
+        action_dist = fwd_out["action_dist"]
+        loss = -tf.math.reduce_mean(action_dist.log_prob(batch[SampleBatch.ACTIONS]))
+        loss_dict = {
```

Comment:
Similar to the torch version return a simple tensor?

Reply:
its gotten complicated to make this work, involving a set of tf.conds. 

I'd prefer if we just enforce that we return a dictionary with a protected keyword. It's a stylistic thing at this point.

Reply:
In your documentation you have said it can be either a tensor or a dict of tensors. That part is a convention that you can set. but for both torch and tf it should be the same. 

On the reason that it got more complicated, isn't this telling us that we need to change something here? What happens if we decorate lower level function? e.g. `compute_gradient()`, `compute_loss()`, `forward_train()` of RLModule?

Reply:
conditionals are a bit tricky to make happen inside of a statically traced context since they aren't themselves traced, or thats what I thought at least. 

I tried using regular if logic and everything seemed to work. solving your problem :) .